bitnami/spark/templates/svc-master.yaml

```diff
@@ -15,35 +17,38 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if and .Values.service.clusterIP (eq .Values.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.service.clusterIP }}
+  {{- end }}
   {{- if and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP)) }}
   loadBalancerIP: {{ .Values.service.loadBalancerIP }}
   {{- end }}
   ports:
-    - port: {{ .Values.service.clusterPort }}
+    - port: {{ .Values.service.ports.clusterPort }}
       targetPort: cluster
       name: cluster
-      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.cluster)) }}
-      nodePort: {{ .Values.service.nodePorts.cluster }}
+      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.clusterPort)) }}
+      nodePort: {{ .Values.service.nodePorts.clusterPort }}
```

Comment:
Since it was already `nodePorts.cluster`, why not using `ports.cluster` instead of renaming everything to a redundant `clusterPort`?

Reply:
You are right, I don't know how I missed this. Thanks!